library/core/class.oauth2.php

```diff
@@ -728,11 +729,19 @@ public function getProfile() {
             'access_token' => $this->accessToken()
         ];
 
+        // Either send the Access Token in the GET request or as an Authorization header, depending on the client workflow.
+        if (val('BearerToken', $provider)) {
+            $defaultParams = [];
+            $profileRequestOptions = array_merge($this->getProfileRequestOptions(), ['Authorization-Header-Message' => 'Bearer '.$this->accessToken()]);
+        } else {
+            $profileRequestOptions = $this->getProfileRequestOptions();
+        }
+
```

Comment:
This all reads very confusedly for what it's doing.

Why let `$defaultParams` be set then unset it? 

Why put `$profileRequestOptions = $this->getProfileRequestOptions()` in an else block rather than just doing that first either way?

You're twisting together logic here that should just be 1 nice if block.